### PR TITLE
openshift master scripts refactor

### PR DESCRIPTION
**Pull request by mjudeikis**
This pr does few things:

* make script idempotent, so it can be rerun when developing
* adds ip addr validation (each time when we call `dig`, `hostname`, we should always validate the output!)
*  splits code in functions so they could be deletes or commented out one by one if needed when developing. 

Next steps would be to remove some of the code. But when code is idempotent we can move agent install to base image and dont worry it will break downstream code (less code orchestration)  

@kwoodson @jim-minter @pweil- @kargakis

### CODE REVIEW

**Comment by mjudeikis** [comment]
I know we want to get rid of this script, but this will enable easier development for now :)

File: parts/openshift/unstable/openshiftmasterscript.sh [hunk 1]
```diff
@@ -53,13 +111,10 @@ oc adm create-bootstrap-policy-file --filename=/etc/origin/master/policy.json
...
-# additional functionality in this script that requires it.  One route to remove
-# this code is to bake this script into the base image, then pass in parameters
-# such as the registry storage account name and key direct from ARM.
-rpm -i https://packages.microsoft.com/yumrepos/azure-cli/azure-cli-2.0.31-1.el7.x86_64.rpm
-
+# configure azure cloud provider
+function configure_cloud_provider() {
 set +x
 . <(sed -e 's/: */=/' /etc/origin/cloudprovider/azure.conf)
 az login --service-principal -u "$aadClientId" -p "$aadClientSecret" --tenant "$aadTenantId" &>/dev/null
```
Comment (mjudeikis) [inline parts/openshift/unstable/openshiftmasterscript.sh]: this one need better error handling as if variables are passed wrong it errors to /dev/null

**Comment by kargakis** [comment]
I would try to keep this in sync with the script in release39 if we want to move forward with this.

**Comment by mjudeikis** [comment]
Agreed. Let first get consensus if people fine with structure and Will change 3.9 accordingly too.

**Comment by jim-minter** [comment]
I don't want to do this.  The master script goes away with HCP.

**Pull request closed without merging**